packages/extension-authenticator-canner/src/lib/authenticator/pat.ts

```diff
@@ -110,11 +110,14 @@ export class CannerPATAuthenticator extends BaseAuthenticator<CannerPATOptions>
   }
   private getCannerUrl(path = '/') {
     const { host, port, ssl = false } = this.options;
-    if (config.isOnKubernetes)
-      return `http://${process.env['WEB_SERVICE_HOST']}${path}`; // for internal usage, we don't need to specify port
+    // the web endpoint is different between connecting from internal or external
+    if (config.isOnKubernetes) return `${process.env['WEB_ENDPOINT']}${path}`;
     else {
       const protocol = ssl ? 'https' : 'http';
-      return `${protocol}://${host}${port ? `:${port}` : ''}${path}`;
+      // our nginx proxy will redirect all request to web service when path starts with /web
```

Comment:
I think the comment could be put in the above condition and simply modify like the below:
```ts
On K8s, we use nginx proxy and it will redirect all requests to web service when the path starts with /web, so we don't need to add the path prefix `/web`.
```
because nginx redirects to `/web` related to the remote environment on k8s.